app.py

```diff
@@ -60,17 +95,70 @@ def show_in_filesystem(self, entity_type, entity_ids):
         with the given entity ids.
         """
         paths = []
-        
+
+        # As the this method relies on the path_cache to find the folders,
+        # we must check we are in sync before continuing
+        self.sgtk.synchronize_filesystem_structure()
+
         for eid in entity_ids:
+
+            if entity_type == "PublishedFile":
+                # Find the publish so that we can get the path back.
+                published_file = self.sgtk.shotgun.find_one("PublishedFile", [["id", "is", eid]], ['code', 'path'])
+                # Resolve the path for the local OS
+                try:
+                    local_path = sgtk.util.resolve_publish_path(self.sgtk, published_file)
+                except Exception as e:
+                    self.log_debug("Publish path couldn't be resolved falling back to context based path; reason: %s" % e)
+                    local_path = None
+
+                if local_path:
+                    if (os.path.isdir(local_path) or os.path.isfile(local_path)):
+                        # If the path is pointing to a valid file or folder then its good to use as is.
+                        paths.append(local_path)
+                        continue
+
+                    # possibly we have an image sequence and therefore its a symbolic path, instead check to see if
```

Comment:
I agree with you, and I've refactored it, however by doing this, it will now perform this hacky sequence check on all paths, not just ones that come from the PublishedFile entities.